components/scroller/scroller.stories.js

```diff
@@ -0,0 +1,118 @@
+import DtScroller from './scroller';
+import BaseScrollerMdx from './scroller.mdx';
+import ScrollerDefault from './scroller_default.story';
+import ScrollerDynamic from '@/components/scroller/scroller_dynamic.story.vue';
+import { createTemplateFromVueFile } from '@/common/storybook_utils';
+import { action } from '@storybook/addon-actions';
+import { CHIP_SIZE_MODIFIERS } from '@';
+
+export const argTypesData = {
+  default: {
+    control: { type: null },
+    description: 'Slot for main content that will be rendered inside the scroller. ' +
+      'Default scoped slot props are:\n\n ' +
+      'item: item being rendered in a view.\n\n' +
+      'index: reflects each items position in the items array\n\n' +
+      'active: whether or not the view is active.',
+    table: {
+      type: {
+        summary: 'VNode',
+      },
+    },
+  },
+
+  // Props
+  items: {
+    control: { type: null },
+  },
+
+  dynamic: {
+    control: { type: null },
+  },
+
+  keyField: {
+    control: { type: null },
+  },
+
+  direction: {
+    control: {
+      type: 'select',
+      options: ['horizontal', 'vertical'],
+    },
+  },
+
+  listTag: {
+    control: { type: 'text' },
+    defaultValue: 'div',
+  },
+
+  itemTag: {
+    control: { type: 'text' },
+    defaultValue: 'div',
+  },
+
+  itemSize: {
+    control: { type: 'number' },
+    defaultValue: 32,
+  },
+
+  minItemSize: {
+    control: { type: null },
+  },
+
+  scrollerWidth: {
+    control: { type: 'number' },
+    defaultValue: 300,
+  },
+
+  scrollerHeight: {
+    control: { type: 'number' },
+    defaultValue: 200,
+  },
+
+  // Events
+  onScrollStart: {
+    control: { type: null },
+    description: 'Emitted when the first item is rendered.',
+  },
+
+  onScrollEnd: {
+    control: { type: null },
+    description: 'Emitted when the last item is rendered.',
+  },
+
+};
+
+export const argsData = {
+  onScrollStart: action('scroll-start'),
+  onScrollEnd: action('scroll-end'),
+};
```

Comment:
The way storybook works here is a little confusing, but normally when you have events you have to do them like this to get them to display in storybook properly. Can see some of our other existing files for examples.

```js
// Action Event Handlers
onScrollStart: {
  table: {
    disable: true,
  },
},
onScrollEnd: {
  table: {
    disable: true,
  },
},

'scroll-start': {
  description: 'Emitted when the first item is rendered.',
  table: {
    type: { summary: 'event' },
  },
},
'scroll-end': {
  description: 'Emitted when the last item is rendered.',
  table: {
    type: { summary: 'event' },
  },
},
```

Reply:
Nice